lib/system/connection_probe.rb

```diff
@@ -8,12 +8,7 @@ class ConnectionProbe < ActiveRecord::Base
 
       class << self
         def connection_config
-          spec = configuration_specification.config.dup
-          if oracle?
-            spec[:password] = ENV.fetch('ORACLE_SYSTEM_PASSWORD') {|key| raise KeyError, "Environment #{key} is mandatory"}
-            spec[:username] = 'SYSTEM'
-          end
-          spec
+          configuration_specification.config
```

Comment:
I suspect this also has to be conditional on the presence of `ORACLE_SYSTEM_PASSWORD`. Because otherwise `rake db:create` may fail if it is run on a clean database with `ORACLE_SYSTEM_PASSWORD` provided.

Reply:
Yes I can. There were no errors. I'm confused... Did you get errors?
And what's the suspicion? That this will be executed before the non-SYSTEM user is created and granted permissions, since we're using `ORACLE_SYSTEM_PASSWORD`?

Reply:
Because if user was not yet created in Oracle, how can probe work?

Reply:
Ah yes. So that was that... The suspicion was really not clear to me before.
If the user is not there, then the connection probe will fail, having the same behaviour as for MySQL and Postgres, if the provided user doesn't exist :v:

Reply:
but then it will not work automatically *unlike* mysql and pgsql in our dev environments

Reply:
while it worked automatically before. Also I went to a great length (see `Rakefile`) to make `db:reset` just work on MySQL on a clean database. I think we deserve DB setup to work for us OOB.

Reply:
Also I think CI failed otherwise.

Reply:
> but then it will not work automatically unlike mysql and pgsql in our dev environments

I reverted the changes on the makefile to include `ORACLE_SYSTEM_PASSWORD` again so nothing changes there. It should be working out of the box as in "the application creates everything and grants everything for you".

Reply:
@akostadinov What do you want me to do here? Add the condition for when `ORACLE_SYSTEM_PASSWORD` is provided so it performs those queries I removed using the SYSTEM user anyway?

Reply:
End goal is that `rails db:reset` would work on a clean database without any non-system users created.

If this is the case right now - then we are ok.

My understanding from looking at the code is that if `ORACLE_SYSTEM_PASSWORD` is provided, then use it to run same query that you changed, no need to run this other complicated query. Because at the time of connection probe the non-system user might be still unavailable. 

Again correct me if I'm wrong. If the above already always works, then I'm fine to keep as is.

Reply:
From all the tests I was able to do, it worked as expected. I'm failing to make it... fail.
Even so, I'm just not confident anymore that this won't fail at some other case I'm not able to understand now or that maybe happened in the past. Gotta say this is quite frustrating...

For the sake of "... I really don't know", I re-added the oracle check to use the SYSTEM user if the password is provided in 749b1e5.

Reply:
let me try locally and will let you know, no need for the check if it works without it

Reply:
> let me try locally

Yeah sure.
And I do agree that, looking just at the code, it makes more sense to use the SYSTEM user in case the password is provided. In any case, we're pretty much keeping the core parts of what's already working today and adding the option to not use the SYSTEM user.